lib/Jsrt/ChakraCore.h

```diff
@@ -179,10 +234,10 @@ JsCreateEnhancedFunction(
 /// <remarks>
 ///     Bootstrap the module loading process by creating a new module record.
 /// </remarks>
-/// <param name="referencingModule">The referencingModule as in HostResolveImportedModule (15.2.1.17). nullptr if this is the top level module.</param>
-/// <param name="normalizedSpecifier">The host normalized specifier. This is the key to a unique ModuleRecord.</param>
-/// <param name="moduleRecord">The new ModuleRecord created. The host should not try to call this API twice with the same normalizedSpecifier.
-///                           chakra will return an existing ModuleRecord if the specifier was passed in before.</param>
+/// <param name="referencingModule">The parent module of the new module - nullptr for a root module.</param>
+/// <param name="normalizedSpecifier">The normalized specifier for the module.</param>
+/// <param name="moduleRecord">The new module record. The host should not try to call this API twice
+///                            with the same normalizedSpecifier.</param>
```

Comment:
Is the behavior we're warning against undefined here or does this crash? The note before your change indicates host would just get the existing ModuleRecord back. If we really don't want host to do this, Jsrt layer could throw an error.

Reply:
I'd like this to be edited to give the existing record back if you supply the same normalised specifier twice, see my suggestion #4740 

At the moment it creates a new record every time it's called - per spec one referencing module/specifier pair (i.e. one normalised specifier) should always lead to only one module record. Currently to implement this correctly the host to maintain their own map of records as this function will just make a new one whatever you supply (see the moduleRecord map in ch as an example.)

Reply:
To clarify the existing comment is just wrong - I'd like it to be right but that would require some work to change the API which is my suggestion I've referenced.

In the mean time editing the comment to document the actual current behaviour seemed appropriate @boingoing

Reply:
Yeah that's fair. Let's keep the comment as it is until we can fix #4740.